src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -131,6 +136,10 @@ def _set_params(
         if include_multi_channel_metrics:
             metric_names += get_multi_channel_template_metric_names()
 
+        # `run` cannot take parameters, so need to find another way to pass this
+        self.delete_existing_metrics = delete_existing_metrics
+        self.metric_names = metric_names
```

Comment:
We should not have self attribute because of reload it do not works.
So everything should be either in self.parms or self.data.
In that case lets put this to self.params.

Reply:
Ok done - we also need to propogate `metrics_to_compute`, which looks a bit more out of place. Alternately we could repeat the code near line 90 of `quality_metric_calculator.py` .